core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/BedrockSetDefaultGameTypeTranslator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019-2023 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.translator.protocol.bedrock.entity.player;
+
+import com.github.steveice10.mc.protocol.data.game.entity.player.GameMode;
+import org.cloudburstmc.protocol.bedrock.packet.SetDefaultGameTypePacket;
+import org.geysermc.geyser.session.GeyserSession;
+import org.geysermc.geyser.translator.protocol.PacketTranslator;
+import org.geysermc.geyser.translator.protocol.Translator;
+
+@Translator(packet = SetDefaultGameTypePacket.class)
+public class BedrockSetDefaultGameTypeTranslator extends PacketTranslator<SetDefaultGameTypePacket> {
+
+    /**
+     * Sets the default game mode for the server via the Bedrock client's "world" menu (given sufficient permissions).
+     */
+    @Override
+    public void translate(GeyserSession session, SetDefaultGameTypePacket packet) {
+        if (session.getOpPermissionLevel() >= 2 || session.hasPermission("geyser.settings.server")) {
+            session.getGeyser().getWorldManager().setDefaultGameMode(session, GameMode.byId(packet.getGamemode()));
+        }
+        // update the client
+        SetDefaultGameTypePacket defaultGameTypePacket = new SetDefaultGameTypePacket();
+        defaultGameTypePacket.setGamemode(packet.getGamemode());
```

Comment:
Can we tell the client to keep its gamemode as what the server has it as? We don't want a client to think its in creative when it shouldn't be.

Reply:
I think it is in the case where the client gamemode is set to "default", since it wouldn't update otherwise

Then again, do we want that update?

Reply:
Okay so apparently
If the Bedrock player's gamemode is set to "default", and the default game type is changed, itll just... change it's own game type too without announcing it to the server. Basically, you are right that we dont need to send the default gamemode back, but we should send a gamemode packet to prevent the bedrock player switching their gamemode without the server knowing if they're in "default gamemode"